pkgs/by-name/op/openthread-border-router/package.nix

```diff
@@ -0,0 +1,136 @@
+{
+  lib,
+  stdenv,
+  nixosTests,
+  nix-update-script,
+  buildNpmPackage,
+  fetchFromGitHub,
+  cmake,
+  ninja,
+  pkg-config,
+  avahi,
+  cjson,
+  jsoncpp,
+  http-parser,
+  libnetfilter_queue,
+  libnfnetlink,
+  systemd,
+  cpputest,
+  gtest,
+  boost,
+  dbus,
+  protobuf,
+  dhcpcd,
+  nodePackages,
+  settingsPath ? "/var/lib/thread",
+  ...
+}:
+
+buildNpmPackage rec {
+  pname = "openthread-border-router";
+  version = "20250618";
+  npmDepsHash = "sha256-7UVfPICyIbHEClpr3p7eDR46OUzS8mVf6P7phnDpVLk=";
+  src = fetchFromGitHub {
+    owner = "openthread";
+    repo = "ot-br-posix";
+    rev = "dcc621476f2586cf4f7a3e2aa93b6db498ac33d3";
+    hash = "sha256-KoXqm8POKVIzgTH36gXghJ4sryc0jBxQdERVnINKozI=";
+    fetchSubmodules = true;
+  };
+  patches = [ ./otbr-fix-vendored.patch ];
+  postPatch = ''
+    rm -rf third_party/{cJSON,http-parser}
+    cp src/web/web-service/frontend/package-lock.json .
+    cp src/web/web-service/frontend/package.json .
+  '';
+
+  nativeBuildInputs = [
+    cmake
+    ninja
+    pkg-config
+    protobuf
+  ];
+  cmakeFlags = [
+    "-DCMAKE_CXX_STANDARD=20"
+
+    "-DINSTALL_SYSTEMD_UNIT=OFF"
+    "-DOT_POSIX_SETTINGS_PATH=\"${settingsPath}\""
+    "-DOT_CHANNEL_MANAGER=ON"
+    #   "-DOT_CHANNEL_MANAGER_CSL=1"
+    #   "-DOT_CSL_RECEIVER=1"
+    #   "-DOT_WAKEUP_COORDINATOR=1"
+    "-DOT_CHANNEL_MONITOR=ON"
+    "-DOT_DNS_CLIENT=ON"
+    "-DOT_DNSSD_SERVER=ON"
+    "-DOT_DNSSD_DISCOVERY_PROXY=ON"
+    
+    "-DOT_PLATFORM_DNSSD=ON"
+    "-DOT_TREL=ON"
+    #    "-DOT_TREL_MANAGE_DNSSD=ON"
+    "-DOT_DNS_UPSTREAM_QUERY=ON"
```

Comment:
A lot of the `OT_` defines are set in [third_party/openthread/CMakeLists.txt](https://github.com/openthread/ot-br-posix/blob/main/third_party/openthread/CMakeLists.txt#L29-L76) and some are copied from their `OTBR_` variant.

- `OT_DNS_CLIENT` is always on
- `OT_DNSSD_SERVER` is set to `${OTBR_DNSSD_DISCOVERY_PROXY}`  
- `OT_DNS_UPSTREAM_QUERY` is set to `${OTBR_DNS_UPSTREAM_QUERY}`
- `OT_PLATFORM_DNSSD` is set to `${OTBR_DNSSD_PLAT}`
- `OT_DNS_UPSTREAM_QUERY` is set to `${OTBR_DNS_UPSTREAM_QUERY}`

Reply:
Good catch!